File: python/easy/uncommon_words_from_two_sentences.py
from collections import Counter
from typing import List


def uncommon_words(s1: str, s2: str) -> List[str]:
    words = s1.split() + s2.split()
    return [word for word in words if words.count(word) == 1]


def uncommon_words_counter(s1: str, s2: str) -> List[str]:
    words = s1.split() + s2.split()
    word_count = Counter(words)
    return [word for word, count in word_count.items() if count == 1]

File: rust/src/easy/uncommon_words_from_two_sentences.rs
use std::collections::HashMap;

pub fn uncommon_words(s1: &str, s2: &str) -> Vec<String> {
    let mut word_count = HashMap::new();

    for word in s1.split_whitespace().chain(s2.split_whitespace()) {
        *word_count.entry(word).or_insert(0) += 1;
    }

    word_count
        .into_iter()
        .filter(|&(_, count)| count == 1)
        .map(|(word, _)| word.to_string())
        .collect()
}
